plugins/by-name/ltex-extra/default.nix

```diff
@@ -43,26 +43,59 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ] "Log level.";
   };
 
-  extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+  extraConfig =
+    cfg:
+    let
+      inherit (config.plugins.lsp.servers) ltex;
+      attachLua = ''
+        require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    in
+    {
+      warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+        {
+          when = !config.plugins.lsp.enable;
+          message = ''
+            You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+            You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+          '';
+        }
+        {
+          # TODO: Added 2025-03-30; remove after 25.05
+          # Warn if someone seem to have an externaly configured servers.ltex
+          when =
+            ltex.enable != true
+            && (
+              ltex.onAttach.function != attachLua
+              || ltex.onAttach.override != false
+              ||
+                [ ] != collect (v: !(isAttrs v || v == null)) (
+                  ltex
+                  // {
+                    extraSettings = null;
+                    package = null;
+                    onAttach = null;
+                  }
```

Comment:
Shouldn't those be inverted? Here, `extraSettings` & co will be overridden. Am I missing something?

Reply:
So I'm trying to find any `defaultNullOpts` that have been modified. This should mask options that are not a `defaultNullOpt` or, as in the case of `onAttach`, expected to be set by this module.

Reply:
To be even more clear, I use `collect` to find all non-null properties of `lsp.server.ltex`. If there are any such cases, i.e. the result `!= []` then the user configuration has changes other than those previously caused by the `ltex-extra`-module.